backend/src/main/java/gov/cdc/usds/simplereport/service/ResultService.java

```diff
@@ -16,6 +35,140 @@
+      Date endDate) {
+    return (root, query, cb) -> {
+      Join<Result, TestEvent> testEventJoin = root.join(Result_.testEvent);
+      Join<TestEvent, Person> personJoin = testEventJoin.join(TestEvent_.patient);
+      Predicate p = cb.conjunction();
+
+      query.orderBy(cb.desc(root.get(AuditedEntity_.createdAt)));
+      query.distinct(true);
+
+      if (facilityId != null) {
+        p =
+            cb.and(
+                p,
+                cb.equal(
+                    testEventJoin.get(BaseTestInfo_.facility).get(IdentifiedEntity_.internalId),
+                    facilityId));
+      } else {
+        final UUID finalOrgId = organizationService.getCurrentOrganization().getInternalId();
+
+        p =
+            cb.and(
+                p,
+                cb.equal(
+                    testEventJoin.get(BaseTestInfo_.organization).get(IdentifiedEntity_.internalId),
+                    finalOrgId));
+      }
+
+      if (patientId != null) {
+        p =
+            cb.and(
+                p,
+                cb.equal(
+                    testEventJoin.get(BaseTestInfo_.patient).get(IdentifiedEntity_.internalId),
+                    patientId));
+      }
+
+      if (result != null) {
+        p = cb.and(p, cb.equal(root.get(Result_.testResult), result));
+      }
+
+      if (role != null) {
+        p = cb.and(p, cb.equal(personJoin.get(Person_.role), role));
+      }
+
+      if (disease != null) {
+        p = cb.and(p, cb.equal(root.get(Result_.disease), disease));
+      }
+
+      if (startDate != null) {
+        p =
+            cb.and(
+                p,
+                cb.or(
+                    cb.and(
+                        cb.isNotNull(testEventJoin.get(BaseTestInfo_.dateTestedBackdate)),
+                        cb.greaterThanOrEqualTo(
+                            testEventJoin.get(BaseTestInfo_.dateTestedBackdate), startDate)),
+                    cb.and(
+                        cb.isNull(testEventJoin.get(BaseTestInfo_.dateTestedBackdate)),
+                        cb.greaterThanOrEqualTo(
+                            testEventJoin.get(AuditedEntity_.createdAt), startDate))));
+      }
+
+      if (endDate != null) {
+        p =
+            cb.and(
+                p,
+                cb.or(
+                    cb.and(
+                        cb.isNotNull(testEventJoin.get(BaseTestInfo_.dateTestedBackdate)),
+                        cb.lessThanOrEqualTo(
+                            testEventJoin.get(BaseTestInfo_.dateTestedBackdate), endDate)),
+                    cb.and(
+                        cb.isNull(testEventJoin.get(BaseTestInfo_.dateTestedBackdate)),
+                        cb.lessThanOrEqualTo(
+                            testEventJoin.get(AuditedEntity_.createdAt), endDate))));
+      }
+
+      return p;
+    };
+  }
+
+  @Transactional(readOnly = true)
+  @AuthorizationConfiguration.RequirePermissionViewAllFacilityResults
+  public Page<Result> getOrganizationResults(
```

Comment:
remove `getOrganizationResults` and use `getFacilityResults` since it can handle when facility is null

Reply:
The two methods are configured with different authorizations requirements:

| Method | Auth |
|--------|--------|
| getFacilityResults | RequirePermissionReadResultListAtFacility |
| getOrganizationResults | RequirePermissionViewAllFacilityResults | 

Is there a clean way of conditionally applying those requirements within a single method, or shall I keep it as-is?

Reply:
I think it's good as is if they have different auth requirements